pkg/probe/probe_builder.go

```diff
@@ -192,8 +191,37 @@ func (pb *ProbeBuilder) DiscoversTarget(targetId string, discoveryClient TurboDi
 		return pb
 	}
 
-	pb.discoveryClientMap[targetId] = discoveryClient
+	pb.targetsToAdd[targetId] = true
+	if pb.discoveryClient != nil {
+		glog.Warningf("Overwriting discovery client %v with %v", pb.discoveryClient, discoveryClient)
+	}
+	pb.discoveryClient = discoveryClient
+
+	return pb
+}
 
+// Set a target for the probe
+func (pb *ProbeBuilder) DiscoversTargetId(targetId string) *ProbeBuilder {
```

Comment:
Is this function being used anywhere in `prometurbo`?

Reply:
Nope. I can drop it if it's not necessary. It just permits you to add additional targets to add at startup without overwriting the discovery client you maybe assigned elsewhere.